### ISSUE TITLE
FYI - `google-cloud-cpp` is a C++ library with a native OpenTelemetry instrumentation

### ISSUE THREAD

**Issue by dbolduc** [open-telemetry/opentelemetry.io#4753]
I am responding to this request from https://opentelemetry.io/docs/languages/cpp/library/#use-natively-instrumented-libraries:

![image](https://github.com/open-telemetry/opentelemetry.io/assets/23088558/7e4e5cc7-8246-4fc4-9fdb-f31e75ebbc30)

The Google Cloud C++ Client Libraries are natively instrumented with OpenTelemetry to collect traces.

Repo:
https://github.com/googleapis/google-cloud-cpp/

Tracing Quickstart:
https://cloud.google.com/cpp/docs/quickstart-open-telemetry

Slightly more detailed quickstart (with pictures of traces!):
https://github.com/googleapis/google-cloud-cpp/tree/main/google/cloud/opentelemetry/quickstart

---

Let me know how to add `google-cloud-cpp` as an integration for `opentelemetry-cpp`. I am happy to help.

**Comment by svrnm** [comment]
Thanks for sharing @dbolduc!

@open-telemetry/cpp-approvers PTAL

**Comment by dbolduc** [comment]
@svrnm nothing has happened. I am not sure `@open-telemetry/cpp-approvers` is a thing.

**Comment by svrnm** [comment]
> @svrnm nothing has happened. I am not sure `@open-telemetry/cpp-approvers` is a thing.

yes, this is a thing, you can not see the team since you are not a member of the github opentelemetry org, a limitation by github.

Note that you can add the library to the registry and set the library to be native to be listed on that page

**Comment by svrnm** [comment]
this can be fixed by adding a file to the registry that includes deatils on `google-cloud-cpp`
